### PR TITLE
feat: add support for Python 3.13

### PR DESCRIPTION
**Pull request by rhatgadkar-goog**

### CODE REVIEW

**Comment by jackwotherspoon** [comment]
@rhatgadkar-goog PR looks great, to get this PR green you will need a couple things:

- this PR will be blocked until #375 is merged which updates asyncpg to 0.30.0 for tests
-  will need to update required checks in branch protection settings to be 3.9 and 3.13 and no longer 3.8 and 3.12

Nice stuff here! 👏

**Comment by rhatgadkar-goog** [comment]
Thanks Jack! How do I update the branch protection settings? I checked out [this page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#editing-a-branch-protection-rule), but wasn't able to find the Settings menu of the repository?

**Comment by enocom** [comment]
You should now have permission to see that page. :wink:

File: .github/workflows/tests.yaml
Comment (enocom) [inline .github/workflows/tests.yaml:40]: s/depcrecated/deprecated/

**Review by enocom — APPROVED** [review]

File: .github/workflows/tests.yaml
Comment (rhatgadkar-goog) [reply .github/workflows/tests.yaml:40]: Not keeping this change in this PR. Just added that commit to check whether the macos tests pass under Python 3.9.

Will add this change as part of https://github.com/GoogleCloudPlatform/alloydb-python-connector/issues/388.

**Review by enocom — APPROVED** [review]
LGTM

**Pull request merged**
